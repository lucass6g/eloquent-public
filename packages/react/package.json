{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
