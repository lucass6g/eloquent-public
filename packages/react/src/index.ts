export {Switch, Case, Default} from './components/conditional-switch'