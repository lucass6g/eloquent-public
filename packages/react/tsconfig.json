{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
