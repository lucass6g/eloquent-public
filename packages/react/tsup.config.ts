import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})