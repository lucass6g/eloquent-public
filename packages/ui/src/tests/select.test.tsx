import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Select } from "..";

describe("Select", () => {
  describe("should match the snapshot", () => {
    it("opened", () => {
      const component = render(
        <Select.Root>
          <Select.Trigger>
            <Select.Value placeholder="select" />
          </Select.Trigger>
        </Select.Root>
      );

      expect(component.baseElement).toMatchSnapshot();
    });

    it("closed", async () => {
      const component = render(
        <Select.Root>
          <Select.Trigger>
            <Select.Value placeholder="select" />
          </Select.Trigger>
          <Select.Content>
            <Select.Group>
              <Select.Label>label</Select.Label>
              <Select.Item value="value-1">value 1</Select.Item>
              <Select.Item value="value-2">value 2</Select.Item>
            </Select.Group>
          </Select.Content>
        </Select.Root>
      );

      const trigger = screen.getByRole("combobox");
      await waitFor(() => userEvent.click(trigger));

      expect(component.baseElement).toMatchSnapshot();
    });
  });

  it("should select correctly", async () => {
    const stateFn = vi.fn();

    render(
      <Select.Root test-id="test" onValueChange={stateFn}>
        <Select.Trigger>
          <Select.Value placeholder="select" />
        </Select.Trigger>
        <Select.Content>
          <Select.Group>
            <Select.Label>label</Select.Label>
            <Select.Item value="value-1">value 1</Select.Item>
            <Select.Item value="value-2">value 2</Select.Item>
          </Select.Group>
        </Select.Content>
      </Select.Root>
    );

    const trigger = screen.getByRole("combobox");

    //select value 1
    await waitFor(() => userEvent.click(trigger));

    const option1 = screen.getByText("value 1");
    await waitFor(() => userEvent.click(option1));
    expect(stateFn).toHaveBeenCalledWith("value-1");

    //select value 2
    await waitFor(() => userEvent.click(trigger));

    const option2 = screen.getByText("value 2");
    await waitFor(() => userEvent.click(option2));
    expect(stateFn).toHaveBeenCalledWith("value-2");
  });

  it("should be disabled", async () => {
    const stateFn = vi.fn();
    render(
      <Select.Root disabled onValueChange={stateFn}>
        <Select.Trigger>
          <Select.Value placeholder="select" />
        </Select.Trigger>
      </Select.Root>
    );

    const trigger = screen.getByRole("combobox");
    await waitFor(() => userEvent.click(trigger));

    expect(trigger).toBeDisabled();
    expect(stateFn).not.toHaveBeenCalledWith();
  });
});
