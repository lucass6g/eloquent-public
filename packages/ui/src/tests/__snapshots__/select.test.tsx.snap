// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Select > should match the snapshot > closed 1`] = `
<body
  data-scroll-locked="1"
  style="pointer-events: none;"
>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <button
      aria-autocomplete="none"
      aria-controls="radix-:r1:"
      aria-expanded="true"
      class="flex items-center justify-between whitespace-nowrap text-sm shadow-sm ring-offset-background placeholder:text-muted-foreground h-9 w-full px-3 py-2 gap-2 bg-transparent disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 focus:outline-none focus:ring-1 focus:ring-ring border border-input rounded-md"
      data-placeholder=""
      data-state="open"
      dir="ltr"
      role="combobox"
      type="button"
    >
      <span
        style="pointer-events: none;"
      >
        select
      </span>
      <svg
        aria-hidden="true"
        class="h-4 w-4 opacity-50"
        fill="none"
        height="15"
        viewBox="0 0 15 15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M4.93179 5.43179C4.75605 5.60753 4.75605 5.89245 4.93179 6.06819C5.10753 6.24392 5.39245 6.24392 5.56819 6.06819L7.49999 4.13638L9.43179 6.06819C9.60753 6.24392 9.89245 6.24392 10.0682 6.06819C10.2439 5.89245 10.2439 5.60753 10.0682 5.43179L7.81819 3.18179C7.73379 3.0974 7.61933 3.04999 7.49999 3.04999C7.38064 3.04999 7.26618 3.0974 7.18179 3.18179L4.93179 5.43179ZM10.0682 9.56819C10.2439 9.39245 10.2439 9.10753 10.0682 8.93179C9.89245 8.75606 9.60753 8.75606 9.43179 8.93179L7.49999 10.8636L5.56819 8.93179C5.39245 8.75606 5.10753 8.75606 4.93179 8.93179C4.75605 9.10753 4.75605 9.39245 4.93179 9.56819L7.18179 11.8182C7.35753 11.9939 7.64245 11.9939 7.81819 11.8182L10.0682 9.56819Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
  <div
    data-radix-popper-content-wrapper=""
    dir="ltr"
    style="position: fixed; left: 0px; top: 0px; transform: translate(0px, 0px); min-width: max-content; --radix-popper-available-width: -20px; --radix-popper-available-height: -20px; --radix-popper-anchor-width: 0px; --radix-popper-anchor-height: 0px; --radix-popper-transform-origin: 0% 0px;"
  >
    <div
      class="relative z-50 max-h-96 min-w-[8rem] overflow-hidden rounded-md border bg-popover text-popover-foreground shadow-md data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 data-[state=open]:animate-in data-[state=open]:fade-in-0 data-[state=open]:zoom-in-95 data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=closed]:zoom-out-95 data-[side=bottom]:translate-y-1 data-[side=left]:-translate-x-1 data-[side=right]:translate-x-1 data-[side=top]:-translate-y-1"
      data-align="start"
      data-side="bottom"
      data-state="open"
      dir="ltr"
      id="radix-:r1:"
      role="listbox"
      style="box-sizing: border-box; display: flex; flex-direction: column; outline: none; --radix-select-content-transform-origin: var(--radix-popper-transform-origin); --radix-select-content-available-width: var(--radix-popper-available-width); --radix-select-content-available-height: var(--radix-popper-available-height); --radix-select-trigger-width: var(--radix-popper-anchor-width); --radix-select-trigger-height: var(--radix-popper-anchor-height); pointer-events: auto;"
      tabindex="-1"
    >
      <style>
        [data-radix-select-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-select-viewport]::-webkit-scrollbar{display:none}
      </style>
      <div
        class="p-1 h-[var(--radix-select-trigger-height)] w-full min-w-[var(--radix-select-trigger-width)]"
        data-radix-select-viewport=""
        role="presentation"
        style="position: relative; flex: 1; overflow: auto;"
      >
        <div
          aria-labelledby="radix-:r5:"
          role="group"
        >
          <div
            class="px-2 py-1.5 text-sm font-semibold"
            id="radix-:r5:"
          >
            label
          </div>
          <div
            aria-labelledby="radix-:r6:"
            aria-selected="false"
            class="relative flex w-full cursor-default select-none items-center rounded-sm py-1.5 pl-2 pr-8 text-sm outline-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50 focus:bg-accent focus:text-accent-foreground"
            data-highlighted=""
            data-radix-collection-item=""
            data-state="unchecked"
            role="option"
            tabindex="-1"
          >
            <span
              class="absolute right-2 flex h-3.5 w-3.5 items-center justify-center"
            />
            <span
              id="radix-:r6:"
            >
              value 1
            </span>
          </div>
          <div
            aria-labelledby="radix-:r7:"
            aria-selected="false"
            class="relative flex w-full cursor-default select-none items-center rounded-sm py-1.5 pl-2 pr-8 text-sm outline-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50 focus:bg-accent focus:text-accent-foreground"
            data-radix-collection-item=""
            data-state="unchecked"
            role="option"
            tabindex="-1"
          >
            <span
              class="absolute right-2 flex h-3.5 w-3.5 items-center justify-center"
            />
            <span
              id="radix-:r7:"
            >
              value 2
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;

exports[`Select > should match the snapshot > opened 1`] = `
<body>
  <div>
    <button
      aria-autocomplete="none"
      aria-controls="radix-:r0:"
      aria-expanded="false"
      class="flex items-center justify-between whitespace-nowrap text-sm shadow-sm ring-offset-background placeholder:text-muted-foreground h-9 w-full px-3 py-2 gap-2 bg-transparent disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 focus:outline-none focus:ring-1 focus:ring-ring border border-input rounded-md"
      data-placeholder=""
      data-state="closed"
      dir="ltr"
      role="combobox"
      type="button"
    >
      <span
        style="pointer-events: none;"
      >
        select
      </span>
      <svg
        aria-hidden="true"
        class="h-4 w-4 opacity-50"
        fill="none"
        height="15"
        viewBox="0 0 15 15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M4.93179 5.43179C4.75605 5.60753 4.75605 5.89245 4.93179 6.06819C5.10753 6.24392 5.39245 6.24392 5.56819 6.06819L7.49999 4.13638L9.43179 6.06819C9.60753 6.24392 9.89245 6.24392 10.0682 6.06819C10.2439 5.89245 10.2439 5.60753 10.0682 5.43179L7.81819 3.18179C7.73379 3.0974 7.61933 3.04999 7.49999 3.04999C7.38064 3.04999 7.26618 3.0974 7.18179 3.18179L4.93179 5.43179ZM10.0682 9.56819C10.2439 9.39245 10.2439 9.10753 10.0682 8.93179C9.89245 8.75606 9.60753 8.75606 9.43179 8.93179L7.49999 10.8636L5.56819 8.93179C5.39245 8.75606 5.10753 8.75606 4.93179 8.93179C4.75605 9.10753 4.75605 9.39245 4.93179 9.56819L7.18179 11.8182C7.35753 11.9939 7.64245 11.9939 7.81819 11.8182L10.0682 9.56819Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
</body>
`;
